stl/inc/utility

```diff
@@ -110,14 +110,28 @@ _NODISCARD constexpr _Ty(min)(initializer_list<_Ty>); // implemented in <algorit
 
 _EXPORT_STD template <class _Ty, size_t _Size, enable_if_t<_Is_swappable<_Ty>::value, int> /* = 0 */>
 _CONSTEXPR20 void swap(_Ty (&_Left)[_Size], _Ty (&_Right)[_Size]) noexcept(_Is_nothrow_swappable<_Ty>::value) {
-    if (&_Left != &_Right) {
-        _Ty* _First1 = _Left;
-        _Ty* _Last1  = _First1 + _Size;
-        _Ty* _First2 = _Right;
-        for (; _First1 != _Last1; ++_First1, ++_First2) {
-            swap(*_First1, *_First2); // intentional ADL
+    if (&_Left == &_Right) {
+        // Precondition violation. N4988 [utility.swap]/7: Effects: As if by swap_ranges.
+        // N4988 [alg.swap]/2: Precondition: The two ranges [first1, last1) and [first2, last2) do not overlap.
+        _CSTD abort();
+    }

```

Comment:
This felt squirrelly and I filed LWG-4165. It seems that the standard allows self-swap for most swappable types, and it's weird to me to disallow self-swap of built-in array or `std::array`.

Reply:
+1. LWG is strongly leaning towards allowing overlap in arguments to `swap_ranges` when one range is a prefix of the other. I'd prefer to let this through and annotate the loosening of the precondition in `std::swap_ranges` and `range::swap_ranges` instead.

Reply:
We don't even call `swap_rangesa` from here. We should annotate here in `swap` specialization implementation.